otherlibs/stdune-unstable/temp.mli

```diff
@@ -20,16 +20,17 @@ val destroy : what -> Path.t -> unit
     itself. *)
 val clear_dir : Path.t -> unit
 
-(** [temp_path ~dir ~prefix ~suffix] generates a temporary path in [dir]. The
-    base name of the temporary file is formed by concatenating [prefix], then a
-    suitably chosen integer number, then [suffix]. Note that the path isn't
-    actually created. *)
+(** [temp_path ~dir ~prefix ~suffix] creates a temporary file in [dir]. The base
+    name of the file is formed by concatenating [prefix], then a suitably chosen
+    integer number, then [suffix]. Note that the file must be created to reserve
+    the name in [dir] and prevent other processes from taking it concurrently.
+    If you need a temporary file that does not exist on disk, you can create a
+    temporary directory and safely use any file name there. *)
```

Comment:
Since it creates a temporary file, can we call it `temp_file`? The name `temp_path` is suggestive of the old behavior - creating a path but not committing to a file or dir.

Reply:
Agreed. I'll do the renaming in a separate feature.